### PR TITLE
Enhance find command

### LINKED ISSUE #101: Enhance Find Command
Find flashcards based on 
- English phrase
- Foreign phrase

### PR DESCRIPTION
**Pull request by sharmainec**
### Description

Find command accepts foreign keywords

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code quality improvements

### How to test

1. Manually test by using the find command (e.g. `FIND f/早`)
2. Will output list of flashcards whose foreign phrase contains the keyword

### Checklist

- [x] I have written testcases to cover all new methods
- [x] I have built and manually tested this code
- [x] I have updated the User Guide
- [x] I have updated the Developer Guide
- [ ] I have updated my individual project portfolio
- [ ] I have included a preview deployment link to the updated project documentation

closes #101

### CODE REVIEW

File: src/main/java/lingogo/commons/util/StringUtil.java [hunk 1]
```diff
@@ -38,6 +38,41 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
...
+                    .anyMatch(preppedWord::equalsIgnoreCase);
+        } else { // allow non-full word match
+            String preppedSentence = sentence;
+            String[] charsInPreppedSentence = preppedSentence.split("(?!^)");
+
+            return Arrays.stream(charsInPreppedSentence)
+                    .anyMatch(preppedWord::equalsIgnoreCase);
+        }
+    }
+
```
Comment (aikenwx) [inline src/main/java/lingogo/commons/util/StringUtil.java:75]: Change this whole segment to `sentence.contains(word)`

File: src/main/java/lingogo/logic/commands/FindCommand.java [hunk 2]
```diff
@@ -4,28 +4,29 @@
...
+    public static final String COMMAND_DESCRIPTION = "Finds flashcard(s) that have a matching phrase";
+    public static final String COMMAND_USAGE = "find PHRASE";
     public static final String COMMAND_EXAMPLES = "find Good Morning";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Finds all flashcards whose english phrases contain any of "
+            + ": Finds all flashcards whose english or foreign phrases contain any of "
             + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
             + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
             + "Example: " + COMMAND_WORD + " hello";
```
Comment (aikenwx) [inline src/main/java/lingogo/logic/commands/FindCommand.java:25]: Allow user to specify find by prefix e/ for EnglishPhrase, f/ for ForeignPhrase

**Review by aikenwx — CHANGES_REQUESTED** [review]
Also don't forget to updated user guide documentation for the updated find command

**Review by kwokieee — CHANGES_REQUESTED** [review]
Thanks for making the modifications! In addition to the comments, perhaps for the tests in `FindCommandTest`, `FindCommandParserTest`, and `FlashcardAppParserTest`, we could also add the case where the user uses the `Find` command with both English and Foreign phrase prefix?

**Pull request merged**